strawberry/extensions/__init__.py

```diff
@@ -1,35 +1,16 @@
-from typing import Any, Dict
-
-from strawberry.types import ExecutionContext, Info
-from strawberry.utils.await_maybe import AwaitableOrValue
-
-
-class Extension:
-    def __init__(self, *, execution_context: ExecutionContext):
-        self.execution_context = execution_context
-
-    def on_request_start(self) -> AwaitableOrValue[None]:
-        """This method is called when a GraphQL request starts"""
-
-    def on_request_end(self) -> AwaitableOrValue[None]:
-        """This method is called when a GraphQL request ends"""
-
-    def on_validation_start(self) -> AwaitableOrValue[None]:
-        """This method is called before the validation step"""
-
-    def on_validation_end(self) -> AwaitableOrValue[None]:
-        """This method is called after the validation step"""
-
-    def on_parsing_start(self) -> AwaitableOrValue[None]:
-        """This method is called before the parsing step"""
-
-    def on_parsing_end(self) -> AwaitableOrValue[None]:
-        """This method is called after the parsing step"""
-
-    def resolve(
-        self, _next, root, info: Info, *args, **kwargs
-    ) -> AwaitableOrValue[None]:
-        return _next(root, info, *args, **kwargs)
-
-    def get_results(self) -> AwaitableOrValue[Dict[str, Any]]:
-        return {}
+from .add_validation_rule import AddValidationRule
+from .base_extension import Extension
+from .disable_validation import DisableValidation
+from .parser_cache import UseParserCache
+from .query_depth_limiter import WithQueryDepthLimiter
+from .validation_cache import UseValidationCache
+
+
+__all__ = [
+    "Extension",
+    "AddValidationRule",
+    "DisableValidation",
+    "UseParserCache",
+    "UseValidationCache",
+    "WithQueryDepthLimiter",
+]
```

Comment:
shall we just export "Extension" here? so we don't endup having a huge list of extensions here in future?

Reply:
I quite like being able to import the extension directly from `strawberry.extensions` though! It's nicer than having to duplicate the name in the import.

Reply:
that's a good point, maybe we can add lazy load in future, to be fair I was more worried about extensions that have optional dependencies 😊

Reply:
Ah didn't think about optional dependencies. `__all__` can be dynamic right?

Reply:
I think `__all__` can stay static :) but since we are using python 3.7+ we can use `__getattr__`  on the module level to lazily import things 😊